src/gt4py/next/program_processors/runners/dace_fieldview/transformations/k_blocking.py

```diff
@@ -0,0 +1,439 @@
+        - The partition must exists (see `partition_map_output()`).
+        """
+        if self.block_dim is None:
+            raise ValueError("The blocking dimension was not specified.")
+        elif self.blocking_size is None:
+            raise ValueError("The blocking size was not specified.")
+
+        map_entry: nodes.MapEntry = self.map_entry
+        map_params: list[str] = map_entry.map.params
+        map_range: subsets.Range = map_entry.map.range
+        block_var: str = self.block_dim
+
+        scope = graph.scope_dict()
+        if scope[map_entry] is not None:
+            return False
+        if block_var not in map_entry.map.params:
+            return False
+        if map_entry.map.schedule == dace.dtypes.ScheduleType.Sequential:
+            return False
+        if map_range[map_params.index(block_var)][2] != 1:
+            return False
+        if self.partition_map_output(map_entry, block_var, graph, sdfg) is None:
+            return False
+
+        return True
+
+    def apply(
+        self,
+        graph: Union[SDFGState, SDFG],
+        sdfg: SDFG,
+    ) -> None:
+        """Creates a blocking map.
+
+        Performs the operation described in the doc string.
+        """
+        outer_entry: nodes.MapEntry = self.map_entry
+        outer_exit: nodes.MapExit = graph.exit_node(outer_entry)
+        outer_map: nodes.Map = outer_entry.map
+        map_range: subsets.Range = outer_entry.map.range
+        map_params: list[str] = outer_entry.map.params
+
+        # This is the name of the iterator we coarsen
+        block_var: str = self.block_dim
+        block_idx = map_params.index(block_var)
+
+        # This is the name of the iterator that we use in the outer map for the
+        #  blocked dimension
+        coarse_block_var = "__coarse_" + block_var
+
+        # Now compute the partitions of the nodes.
+        independent_nodes, dependent_nodes = self.partition_map_output(  # type: ignore[misc]  # Guaranteed to be not `None`.
+            outer_entry, block_var, graph, sdfg
+        )
+
+        # Generate the sequential inner map
+        rng_start = map_range[block_idx][0]
+        rng_stop = map_range[block_idx][1]
+        inner_label = f"inner_{outer_map.label}"
+        inner_range = {
+            block_var: subsets.Range.from_string(
+                f"({coarse_block_var} * {self.blocking_size} + {rng_start}):min(({rng_start} + {coarse_block_var} + 1) * {self.blocking_size}, {rng_stop} + 1)"
+            )
+        }
+        inner_entry, inner_exit = graph.add_map(
+            name=inner_label,
+            ndrange=inner_range,
+            schedule=dace.dtypes.ScheduleType.Sequential,
+        )
+
+        # TODO(phimuell): Investigate if we want to prevent unrolling here
+
+        # Now we modify the properties of the outer map.
+        coarse_block_range = subsets.Range.from_string(
+            f"0:int_ceil(({rng_stop} + 1) - {rng_start}, {self.blocking_size})"
```

Comment:
The start index should not be `0` but `{rng_start}`

Reply:
This is the coarse block range (shitty name) should start from 0, because it iterates through the blocks, we also give it another name.
If you look above at line 157 where the inner range is defined, you see that it starts at the expected location.